### PR TITLE
58 substrate improvements

### PR DESCRIPTION
**Pull request by aalbino2**
@ndaelman-hu could you check if the HKL defs are okay please?

### CODE REVIEW

**Comment by aalbino2** [comment]
ah it's very hard to review this because I branched from anothe rfeature branch

check the class https://github.com/FAIRmat-NFDI/nomad-material-processing/blob/acd5b25c369e24e104d0a0e2c966131c10dcc8a9/src/nomad_material_processing/general.py#L447

**Comment by hampusnasstrom** [comment]
But we can't merge this into main before the feature branch is merged, right?

**Comment by aalbino2** [comment]
indeed, but what are we waiting for to merge that?

File: src/nomad_material_processing/general.py [hunk 1]
```diff
@@ -0,0 +1,1336 @@
...
+            defaultDisplayUnit='deg',
+            label='± Miscut Angle Deviation',
+        ),
+        unit='deg',
+    )
+    hkl_absolut_orientation = Quantity(
+        type=str,
+        description="""Direction where the miscut goes toward.
+        Here the absolut Miller indices, (hkl), denote a normal to the planes
+        in the basis of the primitive reciprocal lattice vectors. """,
```
Comment (ndaelman-hu) [inline src/nomad_material_processing/general.py:484]: I think the intent is good, but it reads a little bit weird:

- the indices eventually denote (families of) planes: that's what the crystallographer cares about.
How does it denote those planes? It specifies a vector orthogonal to those planes in reciprocal space.
In your phrasing, it sounds the opposite way around.
- _normal_: they're just orthogonal, not necessarily normalized. Actually, you can write both (1,1,1) and (2,2,2) to denote different sets of planes. In reciprocal space, the distance between the parallel planes doubles, so in real space if halves. I would add this to the description.

**Comment by hampusnasstrom** [comment]
Nothing I guess, I'm working on getting the nomad-measurements ready.

File: src/nomad_material_processing/general.py [hunk 2]
```diff
@@ -0,0 +1,1336 @@
...
+            component=ELNComponentEnum.NumberEditQuantity,
+            defaultDisplayUnit='deg',
+            label='± Miscut Angle Deviation',
+        ),
+        unit='deg',
+    )
+    hkl_absolut_orientation = Quantity(
+        type=str,
+        description="""Direction where the miscut goes toward.
+        Here the absolut Miller indices, (hkl), denote a normal to the planes
```
Comment (ndaelman-hu) [inline src/nomad_material_processing/general.py:483]: > Here the absolut Miller indices, (hkl)

1. Typo in _absolute_.
2. hkl cover all crystal lattices, but for some (hexagonal and rhombohedral) it's a popular choice to add a fourth index. I haven't worked much with those myself, though I can take a look there too. I think @JosePizarro3 has more experience with hexagonal lattices.

File: src/nomad_material_processing/general.py [hunk 3]
```diff
@@ -0,0 +1,1336 @@
...
+        in the basis of the primitive reciprocal lattice vectors. """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.StringEditQuantity,
+            label='Miscut Absolut Orientation (hkl)',
+        ),
+    )
+    hkl_relative_orientation = Quantity(
+        type=str,
+        description="""Direction where the miscut goes toward. It uses the direct
+        lattice basis instead of the reciprocal lattice. Note that [hkl] is not
```
Comment (ndaelman-hu) [inline src/nomad_material_processing/general.py:493]: > Note that [hkl]

I think you're looking for `{hkl}`... They correspond to the set of `(hkl)` after applying the symmetry operations (of the conventional cell).

Note that you can also indicate directions that using `[hkl]` and direction families with `<hkl>`.

File: src/nomad_material_processing/general.py [hunk 4]
```diff
@@ -0,0 +1,1336 @@
...
+        generally normal to the (hkl) planes, except in a cubic lattice. """,
```
Comment (ndaelman-hu) [inline src/nomad_material_processing/general.py:494]: Similar to my comment above, what matters are the _planes_. These are the ones along which diffraction happens, or along which a surface is cleaved or grown.

**Comment by aalbino2** [comment]
I deleted the branch because it was a nightmare to rebase in main due to conflicts. I'll open a new PR

**Comment by aalbino2** [comment]
I'll try to reopen this

**Review by ndaelman-hu — COMMENTED** [review]
Depending on what you exactly want to express, you'll have to modify the definitions for the Miller indices.

I only took a look at the quantity definitions. Let me know if there's also any normalization that I have to review.

File: src/nomad_material_processing/general.py (hunk 2 above)
Comment (aalbino2) [reply src/nomad_material_processing/general.py:483]: I just had a confirmation from some other guys. They say that when you have 4 indices, the third is pleonastic and can be derived by the 1st, 2nd, and 4th.

For now, we do not insert it, we might think to a subclass for those cells

File: src/nomad_material_processing/general.py (hunk 3 above)
Comment (aalbino2) [reply src/nomad_material_processing/general.py:493]: {hkl} represent a set of equivalent fsmilies, like 001, 010, 100 in cubic systems.

[hkl] is different, it is the family of planes in the direct space

Comment (aalbino2) [reply src/nomad_material_processing/general.py:493]: @ndaelman-hu https://en.wikipedia.org/wiki/Miller_index

check it out

**Pull request closed without merging**